source/expressions.tex

```diff
@@ -986,10 +986,10 @@
 of \tcode{D}. If \tcode{B} is an
 inaccessible\iref{class.access} or
 ambiguous\iref{class.member.lookup} base class of \tcode{D}, a program
-that necessitates this conversion is ill-formed. The result of the
-conversion is a pointer to the base class subobject of the derived class
-object. The null pointer value is converted to the null pointer value of
-the destination type.
+that necessitates this conversion is ill-formed.
+If the source value is a null pointer value, the result is a
```

Comment:
What's the "source value"? Can we reformulate without using this novel term?

Reply:
@jensmaurer In [conv], there's a mixture of "source value", "source expression" and "source prvalue". I suppose that the best replacement here would be "source prvalue".

Reply:
Sounds better, because it aligns with the phrasing in the "If" at the start of the paragraph.

Reply:
@jensmaurer Applied

Reply:
I really don't like using "source prvalue" here. A prvalue is a kind of expression, so this sounds like it's talking about a (static) property of the expression, which it's not.

I think something like "A null pointer value of the source type is converted to a null pointer value of the destination type. A pointer to an object *o* of type `D` is converted to a pointer to the `B` base class subobject of *o*." would work, and fits much better into our pointer value taxonomy in [basic.compound]. Arguably this would need CWG review since it makes it much more clear that such a conversion applied to any other pointer value is undefined.